src/opustrainer/trainer.py

```diff
@@ -199,11 +199,18 @@ def __next__(self):
         assert self._fh is not None
         try:
             # Try to read the next line from our shuffled file
-            line = self._fh.readline()
-            if line == '':
-                raise StopIteration
-            self.line += 1
-            return line
+            while True:
+                line = self._fh.readline()
+                if line == '':
+                    raise StopIteration
+                self.line += 1
+                # assert that the line is well formed, meaning non of the fields is the empty string
+                # If not, try to get a new line from the corpus
+                testline: List[str] = line.rstrip('/r/n').strip().split('/t')
+                for field in testline:
+                    if field == "":
+                        continue
```

Comment:
This continue will only skip the `for field in testline` loop, right? Not go to the next line?